src/Explorer.js

```diff
@@ -1269,42 +1516,111 @@ class Explorer extends React.PureComponent<Props, State> {
-            getDefaultFieldNames={getDefaultFieldNames}
-            getDefaultScalarArgValue={getDefaultScalarArgValue}
-            makeDefaultArg={makeDefaultArg}
-          />
-        ) : null}
-        {mutationFields ? (
-          <RootView
-            fields={mutationFields}
-            operation="mutation"
-            parsedQuery={parsedQuery}
-            onEdit={this._onEdit}
-            schema={schema}
-            getDefaultFieldNames={getDefaultFieldNames}
-            getDefaultScalarArgValue={getDefaultScalarArgValue}
-            makeDefaultArg={makeDefaultArg}
-          />
-        ) : null}
-        {subscriptionFields ? (
-          <RootView
-            fields={subscriptionFields}
-            operation="subscription"
-            parsedQuery={parsedQuery}
-            onEdit={this._onEdit}
-            schema={schema}
-            getDefaultFieldNames={getDefaultFieldNames}
-            getDefaultScalarArgValue={getDefaultScalarArgValue}
-            makeDefaultArg={makeDefaultArg}
-          />
-        ) : null}
+        {relevantDefinitions.map((definition, index) => {
+          const operation:
+            | OperationDefinitionNode
+            | FragmentDefinitionNode = definition;
+          const operationName =
+            operation && operation.name && operation.name.value;
+
+          const operationKind =
+            operation.kind === 'FragmentDefinition'
+              ? 'fragment'
+              : (operation && operation.operation) || 'query';
+
+          const onOperationRename = newName => {
+            const newOperationDef = renameOperation(operation, newName);
+            this.props.onEdit(print(newOperationDef));
+          };
+
+          const fragmentType =
+            operation.kind === 'FragmentDefinition' &&
+            operation.typeCondition.kind === 'NamedType' &&
+            schema.getType(operation.typeCondition.name.value);
+
+          const fragmentFields =
+            fragmentType instanceof GraphQLObjectType
+              ? fragmentType.getFields()
+              : null;
+
+          const fields =
+            operationKind === 'query'
+              ? queryFields
+              : operationKind === 'mutation'
+              ? mutationFields
+              : operationKind === 'subscription'
+              ? subscriptionFields
+              : operation.kind === 'FragmentDefinition'
+              ? fragmentFields
+              : null;
+
+          const fragmentTypeName =
+            operation.kind === 'FragmentDefinition'
+              ? operation.typeCondition.name.value
+              : null;
+
+          return (
+            <RootView
+              key={index}
+              fields={fields}
+              operation={operationKind}
+              name={operationName}
+              definition={definition}
+              onOperationRename={onOperationRename}
+              onTypeName={fragmentTypeName}
+              onEdit={newDefinition => {
```

Comment:
Make onEdit and onRunOperation use class methods.

Reply:
`onEdit` relies on `parsedQuery` (which is inexpensively recoverable in a class method because of the memoization) and `operation` (which is in the local scope of the mapping function and not recoverable). How would you suggest hoisting `operation` out?